src/Graphics/Texture.cpp

```diff
@@ -280,4 +280,30 @@ SDL_BlendMode Texture::blendMode()
     return _blendMode;
 }
 
+bool Texture::blitWithAlpha(Texture* blitMask, int maskOffsetX, int maskOffsetY)
```

Comment:
Nice job. This is good decision to move this code into separate method. We will be able to refactor it later